### PR TITLE
Fix "Request already fulfilled" before the request is actually fulfilled

### PR DESCRIPTION
**Pull request by makramkd**
Pass `max(requestBlockNumber, currBlockNumber)` to the `GetCommitment` call since there is a race condition where the request commitment has been written on chain but not broadcast to the entire network. This could lead us to falsely thinking that a request was "already fulfilled" when in fact we are reading stale on-chain state.

### CODE REVIEW

**Comment by cl-homu** [comment]
:umbrella: The latest upstream changes (presumably 05028563692eab30d032697835e2cd8337fb7eb5) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

File: core/chains/evm/txmgr/models.go [hunk 1]
```diff
@@ -42,6 +43,10 @@ type TransmitCheckerSpec struct {
 	// VRFCoordinatorAddress is the address of the VRF coordinator that should be used to perform
 	// VRF transmit checks. This should be set iff CheckerType is TransmitCheckerTypeVRFV2.
 	VRFCoordinatorAddress common.Address
+
+	// VRFRequestBlockNumber is the block number in which the provided VRF request has been made.
+	// This should be set iff CheckerType is TransmitCheckerTypeVRFV2.
+	VRFRequestBlockNumber *big.Int
```
Comment (Nic0S) [inline core/chains/evm/txmgr/models.go:49]: Do we need the same bugfix in v1 too? Definitely doesn't have to be in this PR / ticket though

**Review by Nic0S — APPROVED** [review]
awesome, sorry for the late review, forgot about this

File: core/chains/evm/txmgr/models.go (hunk 1 above)
Comment (makramkd) [reply core/chains/evm/txmgr/models.go:49]: I think in theory similar thing could occur for V1, but we haven't seen it yet I don't think.

**Review by Nic0S — APPROVED** [review]

**Pull request merged**